internal/services/mssqlmanagedinstance/mssql_managed_instance_resource.go

```diff
@@ -423,22 +486,119 @@ func (r MsSqlManagedInstanceResource) Update() sdk.ResourceFunc {
 				properties.AdministratorLoginPassword = pointer.To(state.AdministratorLoginPassword)
 			}
 
-			metadata.Logger.Infof("Updating %s", id)
+			if metadata.ResourceData.HasChange("azure_active_directory_administrator") {
+				// Need to check if Microsoft AAD authentication only is enabled or not before calling delete, else you will get the following error:
+				// InvalidManagedServerAADOnlyAuthNoAADAdminPropertyName: AAD Admin is not configured,
+				// AAD Admin must be set before enabling/disabling AAD Authentication Only.
+				log.Printf("[INFO] Checking if AAD Administrator exists")
+				aadAdminExists := false
+				resp, err := adminClient.Get(ctx, id.ResourceGroup, id.Name)
+				if err != nil {
+					if !utils.ResponseWasNotFound(resp.Response) {
+						return fmt.Errorf("retrieving the Administrators of %s: %+v", *id, err)
+					}
+				} else {
+					aadAdminExists = true
+				}
+
+				aadAdminRemoved := checkAADAdminRemoved(metadata.ResourceData)
+				if aadAdminRemoved && aadAdminExists {
+					future, err := azureADAuthenticationOnlyClient.Delete(ctx, id.ResourceGroup, id.Name)
+					if err != nil {
+						log.Printf("[INFO] Deletion of AAD Authentication Only failed for %s: %+v", *id, err)
+						return fmt.Errorf("disabling AAD Authentication Only for %s: %+v", *id, err)
+					}
+
+					if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+						return fmt.Errorf("waiting to disable AAD Authentication Only for %s: %+v", *id, err)
+					}
+
+					resp, err := adminClient.Delete(ctx, id.ResourceGroup, id.Name)
+					if err != nil {
+						return fmt.Errorf("removing the AAD Administrator for %s: %+v", *id, err)
+					}
+					if err = resp.WaitForCompletionRef(ctx, client.Client); err != nil {
+						return fmt.Errorf("waiting for removal of AAD Administrator for %s: %+v", *id, err)
+					}
+				}
+
+				aadAdminProps, err := expandMsSqlManagedInstanceAdministrators(state.AzureActiveDirectoryAdministrator)
+				if err != nil {
+					return err
+				}
+				if aadAdminProps != nil {
+					aadAdminChanged := checkAADAdminChanged(metadata.ResourceData)
+					if aadAdminChanged {
+						// if enabling AAD Authentication only, AAD admin must be set first.
+						future, err := adminClient.CreateOrUpdate(ctx, id.ResourceGroup, id.Name, *aadAdminProps)
+						if err != nil {
+							return fmt.Errorf("creating AAD Administrator for %s: %+v", *id, err)
+						}
+						if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+							return fmt.Errorf("waiting for creation of AAD Administrator for %s: %+v", *id, err)
+						}
+
+					}
+					aadAuthOnlyChanged := checkAADAuthOnlyChanged(metadata.ResourceData)
```

Comment:
This does not need to be a separate function, it can be expressed using `ResourceData.HasCHange()`

```suggestion
					if metadata.ResourceData.HasChange("azure_active_directory_administrator.0.azuread_authentication_only_enabled") {
```

Reply:
Fixed.